File: src/model_v4/entities/SBS_ADRTYP.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_AP_ADR } from "./SBS_AP_ADR";
import { SBS_APTYP } from "./SBS_APTYP";


@Entity("SBS_ADRTYP",{schema:"sbsdb"})
@Index("sbsadrtyp_index1",["ADR_TYP",])
@Index("FK2FB5A0379E420B01",["aPTYP_INDEX",])
export class SBS_ADRTYP {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"ADR_INDEX"
        })
    ADR_INDEX:string;


    @Column("varchar",{
        nullable:false,
        length:50,
        name:"ADR_TYP"
        })
    ADR_TYP:string;


    @Column("bigint",{
        nullable:true,
        default: () => "'NULL'",
        name:"FLAG"
        })
    FLAG:string | null;


    @Column("varchar",{
        nullable:true,
        length:200,
        default: () => "'NULL'",
        name:"PARAM"
        })
    PARAM:string | null;



    @ManyToOne(type=>SBS_APTYP, SBS_APTYP=>SBS_APTYP.sBS_ADRTYPs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'APTYP_INDEX'})
    aPTYP_INDEX:SBS_APTYP | null;



    @OneToMany(type=>SBS_AP_ADR, SBS_AP_ADR=>SBS_AP_ADR.aDR_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_AP_ADRs:SBS_AP_ADR[];

}

File: SbsdbServer/sbsdbv4/model/SbsAdrtyp.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_ADRTYP")]
    public partial class SbsAdrtyp
    {
        public SbsAdrtyp()
        {
            SbsApAdr = new HashSet<SbsApAdr>();
        }

        [Column("ADR_INDEX", TypeName = "bigint(20)")]
        public long AdrIndex { get; set; }
        [Required]
        [Column("ADR_TYP", TypeName = "varchar(50)")]
        public string AdrTyp { get; set; }
        [Column("FLAG", TypeName = "bigint(20)")]
        public long? Flag { get; set; }
        [Column("PARAM", TypeName = "varchar(200)")]
        public string Param { get; set; }
        [Column("APTYP_INDEX", TypeName = "bigint(20)")]
        public long? AptypIndex { get; set; }

        [ForeignKey("AptypIndex")]
        [InverseProperty("SbsAdrtyp")]
        public virtual SbsAptyp AptypIndexNavigation { get; set; }
        [InverseProperty("AdrIndexNavigation")]
        public virtual ICollection<SbsApAdr> SbsApAdr { get; set; }
    }
}
